### DISCUSSION TITLE
How can I include an RSS feed from other websites?

### DISCUSSION THREAD

**Discussion by BsNoSi** [datenstrom/community#160] [Ask a question] [answered]
There is a plugin for offering a feed to other sites. Is there a simple solution for _including_ feeds from other pages?

> Additionally to my very rare time for personal investigations I have to admit, that my knowledge about php is poor. I can read it (laborious) and modify it (trial and error…), but this is not enough for creating something (like that) from scratch.

**Comment by markseuffert** [comment]
Sorry, the [feed plugin](https://github.com/annaesvensson/yellow-feed) shows only changes from your own website. To show feeds from other websites you can use any PHP code to display an RSS feed.

**Comment by BsNoSi** [comment]
;) "any PHP code to display" is my challenge. I did not find something that looks reusable to me for creating a plugin (I admit: I didn't spend currently much time on searching). Coding directly within a template is a quite good hint, that makes it easier for me. Thanks! (Success will be documented here)

**Comment by creamyshot** [comment]
:+1:

**Comment by schulle4u** [comment, marked as answer]
I'm using a simple script based on simplexml_load_file function which is already included in php5, so no need for extra classes like magpie or something. 
http://webdesign.weisshart.de/rss2html.php

Explanations are in german, the script itself however is fairly simple to use. A drawback is missing caching functionality, but for me this isn't a problem yet. 
1. Create a new yellow content snippet based on one of the existing snippets, e.g. copy content-main.php to content-home.php, content-about.php or whatever your page is which should parse the feed. 
2. Paste the html2rss function at the beginning of content-home.php. 
3. At the end of the same file, right after the line:  
   `<?php echo $yellow->page->getContent() ?>`  
   you can add your RSS parser like this:  
   `<?php echo rss2html("http://www.domain.tld/feed/",3); ?>`  
   The 3 in this example states that we want to parse the latest 3 items from this feed, you of course may adjust this to your needs. 
4. Now create a file called home.html in your system/themes/templates directory and add the following:  
   `<?php $yellow->snippet("header") ?>`  
   `<?php $yellow->snippet("content-sidebar") ?>`  
   `<?php $yellow->snippet("content-home") ?>`  
   `<?php $yellow->snippet("footer") ?>`  

Thats it. Not an elegant solution, but it works for my purposes. Maybe my next challenge is to transform this into a plugin. 

HTH
Steffen

**Comment by BsNoSi** [comment]
Thank you! That seems to be the thing, I was looking for. I will try this next week.
